decoder/src/main/kotlin/app/redwarp/gif/decoder/Gif.kt

```diff
@@ -318,5 +318,11 @@ class Gif(
             inputStream: InputStream,
             pixelPacking: PixelPacking = PixelPacking.ARGB
         ): Result<Gif> = Parser.parse(inputStream, pixelPacking).map(::Gif)
+
+        fun from(gif: Gif): Result<Gif> = clone(gif).map(::Gif)
+
+        private fun clone(other: Gif): Result<GifDescriptor> {
+            return Result.Success(other.gifDescriptor)
+        }
```

Comment:
Let's simplify to `fun from(gif: Gif): Gif = Gif(gif.gifDescriptor)` : no need to make it a result, as it will always be a success.
(The other method might fail because of IO exception and so on)

Reply:
It makes sense. I've done it.